File: AmaroTest/app/src/main/java/com/test/amaro/amarotest/presentation/core/BaseActivity.java
package com.test.amaro.amarotest.presentation.core;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import com.test.amaro.amarotest.AmaroApplication;
import com.test.amaro.amarotest.dagger.components.ActivityComponent;

public class BaseActivity extends AppCompatActivity implements MvpView {

    private ActivityComponent activityComponent;

    protected final ActivityComponent getActivityComponent() {
        return this.activityComponent;
    }

    @Override
    @SuppressWarnings("UsePropertyAccessSyntax")
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    public BaseActivity() {
        this.activityComponent = AmaroApplication.getApplicationComponent().activityComponent();
    }


}

File: AmaroTest/app/src/main/java/com/test/amaro/amarotest/core/BaseActivity.kt
package com.test.amaro.amarotest.core

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.test.amaro.amarotest.AmaroApplication
import com.test.amaro.amarotest.dagger.components.ActivityComponent

open class BaseActivity : AppCompatActivity() {

  protected var activityComponent: ActivityComponent = AmaroApplication.applicationComponent.activityComponent()

    @Suppress("UsePropertyAccessSyntax")
    override fun onCreate(savedInstanceState: Bundle?) {
      super.onCreate(savedInstanceState)
    }

}
